salt/grains/core.py

```diff
@@ -2746,3 +2746,37 @@ def default_gateway():
         except Exception:
             continue
     return grains
+
+
+def _kernelparams(cmdline):
+    '''
+    Return the kernel boot parameters
+    '''
+
+    try:
+        import shlex
+    except ImportError:
+        return {}
+
+    grains = {'kernelparams': {}}
+
+    for item in shlex.split(cmdline):
+        data = item.split('=')
+
+        value = None
+        if len(data) == 2 and data[1]:
+            value = data[1].strip('"')
+
+        grains['kernelparams'][data[0]] = value
+
+    return grains
+
+
+def kernelparams():
+    try:
+        with salt.utils.files.fopen('/proc/cmdline', 'r') as fhr:
+            cmdline = fhr.read()
+    except IOError:
+        return {}
+
+    return _kernelparams(cmdline)
```

Comment:
Do we need these [multiple returns](https://docs.python-guide.org/writing/style/#returning-values) all the time? And where is the debug logging?

```python
try:
    with salt.utils.files.fopen('/proc/cmdline', 'r') as fhr:
        ret = _kernelparams(fhr.read())
except IOError as err:
    log.debug(err)
    ret = {}

return ret
```

Reply:
I'll simplify. Thanks.